app/validators/array_hostnames_ips_validator.rb

```diff
@@ -0,0 +1,8 @@
+# validates hostnames, IPv4, IPv6 and subnets
+class ArrayHostnamesIpsValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    record.errors.add attribute, _("must contain valid hostnames") unless value.all? do |item|
+      item.to_s.match(URI::HOST) || IPAddr.new(item) rescue nil
```

Comment:
If there's a `rescue nil`, doesn't that mean the check is always going to pass?

I'm also wondering about `item.to_s` on the URI matching but not on the `IPAddr.new`. I don't know what the data type is and it may just be my lack of Rails knowledge. Is this an issue?

Reply:
Good catch that is probably incorrect.

The `to_s` was required because `match` is only defined for String so that the code does not throw error on different classes. To be on the safe side, I will add that too.